tests/components/conversation/test_init.py

```diff
@@ -883,6 +884,8 @@ async def test_turn_on_intent(
     data = {conversation.ATTR_TEXT: sentence}
     if agent_id is not None:
         data[conversation.ATTR_AGENT_ID] = agent_id
+    if conversation_id is not None:
```

Comment:
Instead of adding this check, we can refactor the parameter to be a dict that we always use to update the service data. The two cases would be a dict with the conversation id and an empty dict.

Reply:
Would it be something like this then for `data`?

```json
{
  "conversation_id": { "value": "..." }
}
```

Reply:
Something like this:

```py
pytest.mark.parametrize("conversation_data", [{"conversation_id": "my_new_conversation"} , {}])
```
We can replace `agent_id` similarly.